src/find/matchers/type_matcher.rs

```diff
@@ -0,0 +1,75 @@
+use std::error::Error;
+use std::fs::FileType;
+use walkdir::DirEntry;
+
+use find::matchers::{Matcher, MatcherIO};
+
+/// This matcher checks the type of the file.
+pub struct TypeMatcher {
+    file_type_fn: fn(&FileType) -> bool,
+}
+
+impl TypeMatcher {
+    pub fn new(type_string: &str) -> Result<TypeMatcher, Box<Error>> {
```

Comment:
It's not selecting one between `impl FromStr` and `pub fn new()`. Can have both and actually call `from_str()` in `new()`.

From the [spec](https://doc.rust-lang.org/std/str/trait.FromStr.html) of `FromStr`:

> A trait to abstract the idea of creating a new instance of a type from a string.

IMHO, that's what we are doing here, it's not "an object that can be represented as a string", which sounds more like `impl Display` for me.

Although I don't have strong examples to show the importance of having `impl FromStr`. It's basically conventional stuff, IMHO. So, entirely up to you. :)